internal/graph/check_test.go

```diff
@@ -581,3 +582,150 @@ condition condition1(param1: string) {
 	require.NoError(t, err)
 	require.False(t, resp.Allowed)
 }
+
+func TestUnionCheckFuncReducer(t *testing.T) {
+	ctx := context.Background()
+
+	concurrencyLimit := uint32(10)
+
+	falseHandler := func(context.Context) (*ResolveCheckResponse, error) {
+		return &ResolveCheckResponse{
+			Allowed:            false,
+			ResolutionMetadata: &ResolutionMetadata{},
+		}, nil
+	}
+
+	trueHandler := func(context.Context) (*ResolveCheckResponse, error) {
+		return &ResolveCheckResponse{
+			Allowed:            true,
+			ResolutionMetadata: &ResolutionMetadata{},
+		}, nil
+	}
+
+	t.Run("if no handlers, return allowed:false", func(t *testing.T) {
+		resp, err := union(ctx, concurrencyLimit)
+		require.NoError(t, err)
+		require.False(t, resp.GetAllowed())
+	})
+
+	t.Run("if one handler is truthy and others are falsy, return allowed:true", func(t *testing.T) {
+		resp, err := union(ctx, concurrencyLimit, trueHandler, falseHandler, falseHandler)
+		require.NoError(t, err)
+		require.True(t, resp.GetAllowed())
+	})
+
+	t.Run("if all handlers are falsy, return allowed:false", func(t *testing.T) {
+		resp, err := union(ctx, concurrencyLimit, falseHandler, falseHandler, falseHandler)
+		require.NoError(t, err)
+		require.False(t, resp.GetAllowed())
+	})
+
+	t.Run("if all handlers are falsy, return allowed:false", func(t *testing.T) {
+		resp, err := union(ctx, concurrencyLimit, falseHandler, falseHandler, falseHandler)
+		require.NoError(t, err)
+		require.False(t, resp.GetAllowed())
+	})
+
+	t.Run("if a handler errors but other handler are truthy, return allowed:true", func(t *testing.T) {
+		depthExceededHandler := func(context.Context) (*ResolveCheckResponse, error) {
+			return nil, ErrResolutionDepthExceeded
+		}
+
+		resp, err := union(ctx, concurrencyLimit, depthExceededHandler, trueHandler)
+		require.NoError(t, err)
+		require.True(t, resp.GetAllowed())
+	})
+
+	t.Run("if a handler errors but other handler is falsy, return error and allowed:false", func(t *testing.T) {
+		depthExceededHandler := func(context.Context) (*ResolveCheckResponse, error) {
+			return nil, ErrResolutionDepthExceeded
+		}
+
+		resp, err := union(ctx, concurrencyLimit, depthExceededHandler, falseHandler)
+		require.ErrorIs(t, ErrResolutionDepthExceeded, err))
+		require.False(t, resp.GetAllowed())
+	})
+
+	t.Run("if a handler errors with cycle but other handler is falsy, return allowed:false with a nil error", func(t *testing.T) {
```

Comment:
Please add a corresponding test for the case where one handler returns an ErrCycleDetected while the other returns `allowed: true`. The outcome should be `allowed: true`.